Add options for cleaner excelfile

lsoest opened the pull request:
Hi guerinoni,

I needed an excel file with just the strings and translations:

1. added two command line switches to disable locations/version info. (not sure if this is the way to go)
2. enabled tests on windows machine
3. added debug output for some tests

lsoest committed:
added two commandline options for cleaner output

--no-version no version information is saved to file
--no-location no location information is saved to file

lsoest committed:
tests: added debug output in case of failure

lsoest committed:
clang-format

lsoest committed:
add cli-build and tests for windows

guerinoni commented on .github/workflows/w10.yaml:
```diff
@@ -25,9 +25,19 @@ jobs:
 
       - name: Build GUI
-        shell: bash
+        shell: pwsh
         run: |
           mkdir build-gui
           cd build-gui
           cmake ..
           cmake --build . --parallel
+          ctest --output-on-failure
+
+      - name: Build CLI
+        shell: pwsh
+        run: |
+          mkdir build-cli
+          cd build-cli
+          cmake -DBUILD_CLI:BOOL=ON ..
+          cmake --build . --parallel
+          ctest --output-on-failure
```
Many thanks, I don't have w10 :)

guerinoni reviewed (commented):
Seems all good! Can you add some documentation in Readme about this new parameters, how use it from CLI and so on? :)

lsoest committed:
install required dlls for windows

lsoest committed:
extend readme

- added example for command line switches
- added section to build and install on windows

guerinoni reviewed (approved):
LGTM! :)
